tests/pos/T2091.hs

```diff
@@ -0,0 +1,21 @@
+{-# LANGUAGE KindSignatures #-}
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE TypeOperators #-}
+{-# LANGUAGE GADTs #-}
+
+import Prelude (Bool(..))
```

Comment:
@nikivazou, should this tests run in CI? They would need to be added to `tests.cabal` if so.

Reply:
... and they may have to start with `module T2091 where` for compilation to succeed.

Reply:
Nope, just did it here: https://github.com/ucsd-progsys/liquidhaskell/pull/2107